src/pip/_internal/utils/misc.py

```diff
@@ -199,18 +198,6 @@ def ask(message: str, options: Iterable[str]) -> str:
             return response
 
 
-def ask_input(message: str) -> str:
```

Comment:
we use this function only for prompting username for keyring. So i removed them too.